sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1003,15 +1229,16 @@ public boolean advance() throws IOException {
 
           curRecord = null; // user coders below might throw.
 
-          // apply user coders. might want to allow skipping records that fail to decode.
+          // apply user deserializers.
+          // might want to allow skipping records that fail to deserialize.
```

Comment:
Silently skipping bad records is silent data loss. A better solution for this case is to have a sort of a "dead-letter" side output, containing raw records that could not be deserialized. That's outside the scope of this PR though.